pkg/dependency/credhelper/cred_helper_binary.go

```diff
@@ -117,79 +149,121 @@ func (bin *credhelperbin) fullInstallPath() string {
 // Installed checks if the credential helper already exists in the specified
 // folder and checks if the hash of the installed binary is correct.
 func (bin *credhelperbin) Installed() bool {
-	dirExists, err := afero.DirExists(bin.fs, bin.hcfg.installFolder)
-	if err != nil {
-		bin.l.Errorf("failed to get status of credential helper directory: %v", err)
-		return false
-	}
-	if !dirExists {
-		return false
-	}
-	fileExists, err := afero.Exists(bin.fs, bin.fullInstallPath())
-	if err != nil {
-		bin.l.Errorf("failed to get status of credential helper binary: %v", err)
-		return false
-	}
-	if !fileExists {
-		return false
-	}
-	file, err := bin.fs.Open(bin.fullInstallPath())
-	if err != nil {
+	if binInstalled, err := bin.binaryInstalled(); err != nil {
 		bin.l.Error(err)
 		return false
+	} else if !binInstalled {
+		return false
 	}
-	defer file.Close() //nolint:errcheck // closing the file
-	hash, err := digest.FromReader(file)
-	if err != nil {
+
+	if cfgInstalled, err := bin.configFileInstalled(); err != nil {
 		bin.l.Error(err)
 		return false
-	}
-	if strings.Compare(hash.String(), bin.hcfg.hash) != 0 {
-		bin.l.Info("Hash of the installed credential helper binary does not match")
-		err := bin.fs.Remove(bin.fullInstallPath())
-		if err != nil {
-			bin.l.Error(err)
-		}
+	} else if !cfgInstalled {
 		return false
 	}
+
 	return true
 }
 
 // Install installs and configures the specified credential helper.
 func (bin *credhelperbin) Install() error {
-	if bin.helper == "" {
-		return nil
-	}
-	if strings.Compare(bin.helper, bin.credHelperConfigName()) != 0 {
-		return nil
-	}
-	credHelperName := strings.ReplaceAll(bin.credHelperConfigName(), "-login", "")
-	bin.l.Infof("Installing %s credential helper", credHelperName)
-	mkdirCmd := bin.cmdCreator.Create("mkdir", "-p", bin.hcfg.installFolder)
-	_, err := mkdirCmd.Output()
+	binInstalled, err := bin.binaryInstalled()
 	if err != nil {
-		return fmt.Errorf("error creating installation directory %s, err: %w", bin.hcfg.installFolder, err)
+		return err
 	}
 
-	curlCmd := bin.cmdCreator.Create("curl", "--retry", "5", "--retry-max-time", "30", "--url",
-		bin.hcfg.credHelperURL, "--output", bin.fullInstallPath())
+	if !binInstalled {
+		if bin.helper == "" {
+			return nil
+		}
+		if bin.helper != bin.credHelperConfigName() {
+			return nil
+		}
+		credHelperName := strings.ReplaceAll(bin.credHelperConfigName(), "-login", "")
+		bin.l.Infof("Installing %s credential helper", credHelperName)
+		mkdirCmd := bin.cmdCreator.Create("mkdir", "-p", bin.hcfg.installFolder)
```

Comment:
nit: Maybe its easier to `bin.fs.MkdirAll(bin.hcfg.installFolder, 0o700)` since bin already has the fs dependency.

Reply:
yep, fixed in latest revision